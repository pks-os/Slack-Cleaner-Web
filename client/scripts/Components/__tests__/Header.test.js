import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../Header';

const AVATAR =
  'https://secure.gravatar.com/avatar/f2d55f35afbc3f607e34504f5a2d7d88.jpg?s=192&d=https%3A%2F%2Fa.slack-edge.com%2F7fa9%2Fimg%2Favatars%2Fava_0006-192.png';
const NAME = 'George Jetson';

describe('<Header />', () => {
  it('Renders Correctly', () => {
    const header = renderer.create(<Header />).toJSON();
    expect(header).toMatchSnapshot();
  });

  it('renders a loggedIn Header with a name and an avatar', () => {
    const header = renderer
      .create(<Header name={NAME} avatar={AVATAR} isLoggedIn />)
      .toJSON();
    expect(header).toMatchSnapshot();
  });

  it('does not render a Header component in the loggedIn State with no other props provided', () => {
    const header = renderer.create(<Header isLoggedIn />).toJSON();
    expect(header).toMatchSnapshot();
  });

  it('does not render a Header component in the loggedIn State when other values are provided', () => {
    const header = renderer
      .create(<Header name={NAME} avatar={AVATAR} />)
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
