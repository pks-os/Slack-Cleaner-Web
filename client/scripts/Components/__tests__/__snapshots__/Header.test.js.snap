// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders Correctly 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__Logo"
  >
    Slack Cleaner
  </h1>
  <div
    className="Header__Meta"
  >

    <a
      className="LoginButton"
      href="auth/slack"
    >
      <img
        alt="Sign in with Slack"
        height="40"
        src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
        srcSet="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
        width="172"
      />
    </a>
  </div>
</header>
`;

exports[`<Header /> does not render a Header component in the loggedIn State when other values are provided 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__Logo"
  >
    Slack Cleaner
  </h1>
  <div
    className="Header__Meta"
  >

    <a
      className="LoginButton"
      href="auth/slack"
    >
      <img
        alt="Sign in with Slack"
        height="40"
        src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
        srcSet="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
        width="172"
      />
    </a>
  </div>
</header>
`;

exports[`<Header /> does not render a Header component in the loggedIn State with no other props provided 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__Logo"
  >
    Slack Cleaner
  </h1>
  <div
    className="Header__Meta"
  >

    <a
      className="LoginButton"
      href="auth/slack"
    >
      <img
        alt="Sign in with Slack"
        height="40"
        src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
        srcSet="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
        width="172"
      />
    </a>
  </div>
</header>
`;

exports[`<Header /> renders a loggedIn Header with a name and an avatar 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__Logo"
  >
    Slack Cleaner
  </h1>
  <div
    className="Header__Meta"
  >

    <p
      className="Header__Name"
    >
      Hey
      George Jetson

    </p>
    <img
      alt="George Jetson"
      className="Header__Avatar"
      src="https://secure.gravatar.com/avatar/f2d55f35afbc3f607e34504f5a2d7d88.jpg?s=192&d=https%3A%2F%2Fa.slack-edge.com%2F7fa9%2Fimg%2Favatars%2Fava_0006-192.png"
    />
    <a
      className="Button"
      href="api/logout"
    >
      Logout
    </a>
  </div>
</header>
`;
